components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -272,7 +281,29 @@ private void checkPasswordValidity(UserStoreException e) throws IdentityRecovery
 
             if (cause instanceof PolicyViolationException) {
                 throw IdentityException.error(IdentityRecoveryClientException.class,
-                        IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_POLICY_VIOLATION.getCode(), cause.getMessage(), e);
+                        IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_POLICY_VIOLATION.getCode(),
+                        cause.getMessage(), e);
+            }
+
+            if (StringUtils.isNotBlank(e.getMessage()) && StringUtils.indexOfAny(e.getMessage(),
+                    pwdPatternViolations) >= 0) {
+                UserStoreManager userStoreManager;
+                try {
+                    userStoreManager = IdentityRecoveryServiceDataHolder.getInstance().getRealmService().
+                            getTenantUserRealm(tenantId).getUserStoreManager();
+                } catch (UserStoreException e1) {
```

Comment:
Lets use a meaningful variable name for e1 and we should use the same when throwing the exception